linode/nodepool/framework_models.go

```diff
@@ -0,0 +1,161 @@
+package nodepool
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/linode/linodego"
+	"github.com/linode/terraform-provider-linode/v2/linode/helper"
+)
+
+type NodePoolModel struct {
+	ID         types.String              `tfsdk:"id"`
+	ClusterID  types.Int64               `tfsdk:"cluster_id"`
+	Count      types.Int64               `tfsdk:"node_count"`
+	Type       types.String              `tfsdk:"type"`
+	Tags       types.Set                 `tfsdk:"tags"`
+	Nodes      types.List                `tfsdk:"nodes"`
+	Autoscaler []NodePoolAutoscalerModel `tfsdk:"autoscaler"`
+}
+
+type NodePoolAutoscalerModel struct {
+	Min types.Int64 `tfsdk:"min"`
+	Max types.Int64 `tfsdk:"max"`
+}
+
+type NodePoolNodeModel struct {
+	ID         types.String `tfsdk:"id"`
+	InstanceID types.Int64  `tfsdk:"instance_id"`
+	Status     types.String `tfsdk:"status"`
+}
+
+func flattenLKENodePoolLinode(node *linodego.LKENodePoolLinode) (*basetypes.ObjectValue, diag.Diagnostics) {
```

Comment:
I think we can drop the pointer here for the `node` argument since it isn't optional, which should also fix the gosec error 👍

Reply:
fixed